### PR TITLE
Kubernetes: explicitly install deps during provisioning

### PR DESCRIPTION
**Pull request by jackfrancis**
**What this PR does / why we need it**: Outcome of this change:

Before:

```
azureuser@k8s-master-20389576-0:~$ sudo dpkg-query -l nfs*
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                          Version             Architecture        Description
+++-=============================-===================-===================-===============================================================
un  nfs-common                    <none>              <none>              (no description available)
```

After:

```
azureuser@k8s-master-14035839-0:~$ sudo dpkg-query -l nfs*
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                          Version             Architecture        Description
+++-=============================-===================-===================-===============================================================
un  nfs-client                    <none>              <none>              (no description available)
ii  nfs-common                    1:1.2.8-9ubuntu12.1 amd64               NFS support files common to client and server
un  nfs-kernel-server             <none>              <none>              (no description available)
```

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```
Kubernetes: explicitly install nfs-common during provisioning
```

### CODE REVIEW

**Comment by jackfrancis** [comment]
@andyzhangx FYI

File: parts/k8s/kubernetescustomscript.sh [hunk 1]
```diff
@@ -153,8 +153,11 @@ function installEtcd() {
     retrycmd_if_failure 10 1 5 sudo etcdctl member update $MEMBER ${ETCD_PEER_URL} || exit $ERR_ETCD_CONFIG_FAIL
 }
 
+function installDeps() {
+    apt_get_install 20 30 180 apt-transport-https ca-certificates iptables iproute2 socat util-linux mount ebtables ethtool init-system-helpers nfs-common ceph-common conntrack glusterfs-client ipset jq || exit $ERR_APT_INSTALL_TIMEOUT
```
Comment (andyzhangx) [inline parts/k8s/kubernetescustomscript.sh]: some packages are already installed in base image, install existing packages won't fail in this command, right?

**Comment by karataliu** [comment]
Fixes #3022

File: parts/k8s/kubernetescustomscript.sh (hunk 1 above)
Comment (andyzhangx) [reply parts/k8s/kubernetescustomscript.sh]: and would you also add package list source in comment:
`https://github.com/kubernetes/kubernetes/blob/master/build/debian-hyperkube-base/Dockerfile#L25-L44`

Comment (jackfrancis) [reply parts/k8s/kubernetescustomscript.sh]: To your 1st question, correct, if a package is already installed the command won’t exit non-zero

**Comment by andyzhangx** [comment]
@jackfrancis I am also fixing the flexvolume issue in AKS, it requires `jq` package, you will get following error when install `jq` on one agent node directly:
```
azureuser@aks-nodepool1-26705064-0:~$ sudo apt install jq -y
Reading package lists... Done
Building dependency tree
Reading state information... Done
You might want to run 'apt-get -f install' to correct these:
The following packages have unmet dependencies:
 jq : Depends: libonig2 (>= 5.9.5) but it is not going to be installed
 linux-azure : Depends: linux-headers-azure (= 4.13.0.1018.19) but 4.13.0.1016.17 is to be installed
               Depends: linux-tools-azure (= 4.13.0.1018.19) but 4.13.0.1016.17 is to be installed
               Depends: linux-cloud-tools-azure (= 4.13.0.1018.19) but 4.13.0.1016.17 is to be installed
E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).
```

by running `apt-get -f install` and then run `apt install jq -y`, `jq` package will finally installed.
So I would suggest modify your installation scripts?

**Comment by feiskyer** [comment]
@andyzhangx apt-get update should run first before install.

**Comment by andyzhangx** [comment]
`apt update` does not work on my side, @jackfrancis you may double check whether `jq` is installed after AKS upgrade with this PR, I doubt there could be issue there.
```
azureuser@aks-nodepool1-26705064-1:~$ sudo apt install jq -y
Reading package lists... Done
Building dependency tree
Reading state information... Done
You might want to run 'apt-get -f install' to correct these:
The following packages have unmet dependencies:
 jq : Depends: libonig2 (>= 5.9.5) but it is not going to be installed
 linux-azure : Depends: linux-headers-azure (= 4.13.0.1018.19) but 4.13.0.1016.17 is to be installed
               Depends: linux-tools-azure (= 4.13.0.1018.19) but 4.13.0.1016.17 is to be installed
               Depends: linux-cloud-tools-azure (= 4.13.0.1018.19) but 4.13.0.1016.17 is to be installed
E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).
azureuser@aks-nodepool1-26705064-1:~$ sudo apt update -y
Hit:1 http://azure.archive.ubuntu.com/ubuntu xenial InRelease
Get:2 http://azure.archive.ubuntu.com/ubuntu xenial-updates InRelease [109 kB]
Get:3 http://azure.archive.ubuntu.com/ubuntu xenial-backports InRelease [107 kB]
Hit:4 http://security.ubuntu.com/ubuntu xenial-security InRelease
Hit:5 https://aptdocker.azureedge.net/repo ubuntu-xenial InRelease
Get:6 http://azure.archive.ubuntu.com/ubuntu xenial-updates/main amd64 Packages [785 kB]
Get:7 http://azure.archive.ubuntu.com/ubuntu xenial-updates/universe amd64 Packages [632 kB]
Fetched 1,633 kB in 0s (3,008 kB/s)
Reading package lists... Done
Building dependency tree
Reading state information... Done
21 packages can be upgraded. Run 'apt list --upgradable' to see them.
azureuser@aks-nodepool1-26705064-1:~$ sudo apt install jq -y
Reading package lists... Done
Building dependency tree
Reading state information... Done
You might want to run 'apt-get -f install' to correct these:
The following packages have unmet dependencies:
 jq : Depends: libonig2 (>= 5.9.5) but it is not going to be installed
 linux-azure : Depends: linux-headers-azure (= 4.13.0.1018.19) but 4.13.0.1016.17 is to be installed
               Depends: linux-tools-azure (= 4.13.0.1018.19) but 4.13.0.1016.17 is to be installed
               Depends: linux-cloud-tools-azure (= 4.13.0.1018.19) but 4.13.0.1016.17 is to be installed
E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).
```

**Comment by jackfrancis** [comment]
@andyzhangx Thanks for the call-out.

Just verified jq installation is working on a newly built cluster:

```
azureuser@k8s-master-64268538-0:~$ which jq
/usr/bin/jq
```

By the way, here are the commands we run for that `apt_get_install` func:

https://github.com/Azure/acs-engine/blob/master/parts/k8s/kubernetesprovisionsource.sh#L74

**Pull request merged**